Update assignment V 0.2

JetsadaBinaibrohim-Doppiotech opened the pull request.

JetsadaBinaibrohim-Doppiotech committed:
Update assignment V 0.2

JetsadaBinaibrohim-Doppiotech committed:
Update Assignment -> 0.3

    Change Structure Code

massarinbdoppio commented on Testcase/tc.robot:
```diff
@@ -0,0 +1,47 @@
+***Test Cases***
+Test Case - TC01
+    ${user_TOKEN}    ${user_ID}=               login_keyword.Verify Login API Can POST On Data And Get Token
+    ...              Wulin_taindi@gmail.com    
+    ...              Wulin12* 
```
อันนี้ก็มีตัวแปรนี่หน่า ทำไมถึง hard code ไว้คับ
มีเหตุผลไหมคับเพราะพี่น่า comment ไป 2 ครั้งว่าไม่ให้ hard code นะ

massarinbdoppio commented on Testcase/tc.robot:
```diff
@@ -0,0 +1,47 @@
+
+    ${search_response}=    search_page.Verify Search Page Load    
+    ...                    ${user_TOKEN}                          
+    ...                    Doppee phone
```
ทำไมไม่ใช้อันนี้คะ  test_pd_name 
กับอีกเหตุผลคือ ไม่เอา hard code คับ

massarinbdoppio commented on Testcase/tc.robot:
```diff
@@ -0,0 +1,47 @@
+    ...                     ${test_cardname}                         
+    ...                     ${test_cardtype}                         
+    ...                     ${test_cardbank}                         
+    ...                     Doppio                                   
+    ...                     Tech                                     
+    ...                     The office                               
+    ...                     0123456789                               
```
ไม่เอา hard code คับ

massarinbdoppio commented on Keywords/commom_keyword.robot:
```diff
@@ -0,0 +1,32 @@
+***Keywords***
+Verify Login API Can POST On Data And Get Token
```
เก็บไว้ทำไมคับ

massarinbdoppio commented on Keywords/Page/payment_page.robot:
```diff
@@ -0,0 +1,55 @@
+*** Keywords ***
+Verify Payment Page Load
```
เราไม่ได้ทำอะไรกับ Payment Page เลยค่ะ เราจะเช็คได้ยังไงว่า  Payment Page load อ่ะ 
ชื่อไม่ตรงกับสิ่งที่ทำคับ แก้ชื่อ
ชื่อ api ไม่ช่าย payment info มันคือ order submit ช่ายป่าว จิงๆ มันก็ตรงๆ ตัวน้า

massarinbdoppio commented on Keywords/Page/payment_page.robot:
```diff
@@ -0,0 +1,55 @@
+    Should Be Equal    ${post_resp.json()['status']}      success
+    Should Be Equal    ${post_resp.json()['type']}        order_submit
+
+    Log To Console    Payment Page Valid ✅
```
payment page valid ?? ไปเช็คหน้า page มาหรือป่าว .. ก็ไม่ช่ายถูกไหม ปรับคำใหม่คับ

massarinbdoppio commented on Keywords/Page/payment_page.robot:
```diff
@@ -0,0 +1,55 @@
+    ${payment_ORDERID}=        Set Variable    ${post_resp.json()['order_id']}
+    ${payment_STATUS}=         Set Variable    ${post_resp.json()['status']}
+    ${payment_TYPE}=           Set Variable    ${post_resp.json()['type']}
+    ${total_PRICEmultiply}=    Evaluate        ${original_price} * ${product_qty}
```
อันนี้เราต้องคำนวนเองเหรอคะ เพื่อเอาไปเช็คเหรอ สงสัยเฉยๆ ว่ามันต้องได้ค่านี้ตั้งแต่ตอนส่งไป order_submit ป่าวนะ

JetsadaBinaibrohim-Doppiotech replied:
ค่าอันนี้เป็นค่าจากหลังจากกด Order Submit Payment แล้วครับ ตัว product_qty มันเป็นค่าที่อยู่ใน payload request ผมเลยสร้างตัวแปรกำหนดค่า ให้ product_qty แล้วเอามาใช้คำนวณเลยครับ

ก็คือ product_qty จำเป็นต้องใส่ค่าก่อน แล้วพอใส่ค่าก็เอาค่าที่ใส่มาคำนวณกับ original_price อีกทีครับ

massarinbdoppio replied:
โอเค เข้าใจ คือ พอเป็น api ต้องคำนวน total เอง

massarinbdoppio commented on Testcase/tc.robot:
```diff
@@ -0,0 +1,47 @@
+    ...                    ${user_TOKEN}                          
+    ...                    Doppee phone
+
+    ${total_PRICEminus}    ${product_ID}=        search_feature.Verify Search Product Data    
```
total_PRICEminus คือราคาอะไรนะ ราคาลบอะไรสักอย่าง จากชื่อ หรือมันคือราคาหลังหักส่วนลดงี้เหรอ

JetsadaBinaibrohim-Doppiotech replied:
เป็นราคาที่ได้หลังจาก Login แล้วได้ส่วนลดครับพี่ ตอนแรกผมคำนวณไว้แล้วแก้เปลี่ยนเป็นดึงค่ามาใช้แทนแล้วครับ

massarinbdoppio commented on Keywords/Page/payment_page.robot:
```diff
@@ -0,0 +1,55 @@
+    ...                 phone=${phone}
+    ...                 email=${email}                      
+    ...                 product_id=${product_id}            
+    ...                 original_price=${original_price}
```
original_price  ต่างกับ total_PRICEminus ยังไงนะ อธิบายหน่อย

JetsadaBinaibrohim-Doppiotech replied:
ตัว original_price ที่ผมคิดตอนแรกคือราคาที่ผู้ใช้ยังไม่ได้ Login ครับ ส่วน total_PRICEminus คือเอาราคาต้นมาลบกับส่วนลดครับ
ซึ่งตรงนี้จริงๆคือตัวเดียวกันครับ ผมทำนอกเหนือโจทย์ ตอนนี้แก้ไขแล้วครับ

massarinbdoppio commented on Keywords/Page/check_page.robot:
```diff
@@ -0,0 +1,19 @@
+***Keywords***
+Verify Check Order Page Load
```
แก้ชื่อ มันไม่ใช่ page load 
กับมันไม่ได้ verify อะไรเลยป่ะคะ มันแค่ get ค่าออกมา return กลับไปให้เฉยๆ

massarinbdoppio commented on Testcase/tc.robot:
```diff
@@ -0,0 +1,47 @@
+    ...                  ${product_child}
+
+    Log To Console       Order ID Check Create Lasted : ${formattedMAX}    
+    Log To Console       Price Product Amount Qty : ${total_PRICEmultiply}
```
Price Product Amount Qty ประโยคนี้งงคับ แต่เข้าใจว่าจิลต้องการเขียนว่า ราคารวม ตามจำนวนที่สั่งซื้องี้ป่ะคะ
ซึ่งพี่น่าว่า จิงๆ ใช้ Total price เลยได้นะ
Qty คือตัวย่อของ Quantity ที่แปลว่า จำนวน ซึ่งคือคำเดียวกันกับ Amount คับ

อ่ะ อันนี้คับ
ถ้าอยากได้แบบที่จิลเขียน เขียนแบบนี้ได้ Total Price (product x amount) หรือ Product price with qty แบบนี้ได้มั้งค่ะ

JetsadaBinaibrohim-Doppiotech committed:
Update assignment V 0.4

    Change keywords and variables name
    Change hard code
    Remove commom_keyword.robot
    Edit ${total_PRICEminus}->${product_PRICE} search_feature.robot

massarinbdoppio reviewed (approved):
พอได้แล้วค่ะ จิงๆ ส่วนที่ยังติดอยู่คือการตั้งชื่อ keyword ซึ่งมองว่า สามารถปรับปรุงในงานถัดไปได้
ส่วนของ logic การทำงาน สามารถแยก login  page ได้ดีขึ้นแล้ว
